### PR TITLE
Fix for #8249

### PR DESCRIPTION
**Pull request by aafemt**

### CODE REVIEW

**Comment by romansimakov** [comment]
Let's please write more details in commit messages. A reference to a ticket is not enough IMO. Ticket describes WHAT is fixed but a commit message should describe HOW it was fixed.

File: src/isql/isql.epp [hunk 1]
```diff
@@ -9597,33 +9597,24 @@ static const char* charset_to_string(unsigned charset)
  **************************************/
 	static Firebird::GlobalPtr<Firebird::GenericMap<Firebird::Pair<Firebird::Right<unsigned, string> > > > csMap;
 
-	charset = TTYPE_TO_CHARSET(charset);
```
Comment (asfernandes) [inline src/isql/isql.epp:9600]: What is the relation of this isql changes with #8249 ?

Comment (aafemt) [reply src/isql/isql.epp:9600]: QA team and me should have a way to check field collation to make sure that #8249 really fixed and result of the `CAST()` has the right collation. The simplest way - to print collation in SQLDA along to the charset. That's why this change.

**Comment by aafemt** [comment]
> commit message should describe HOW it was fixed.

It is written in code: `CastNode` now sends the right collation name instead of `NULL` to `DDL_resolve_intl_type()`.

File: src/isql/isql.epp (hunk 1 above)
Comment (asfernandes) [reply src/isql/isql.epp:9600]: First, ISQL showed the charset/collation id, so it was identifiable.
You want to display also the collation name, but this change makes it incorrectly. Collation ids, different from charset ids, are far from be stable between different databases. I suggest to revert this part of the patch.

**Review by asfernandes — CHANGES_REQUESTED** [review]

**Pull request merged**